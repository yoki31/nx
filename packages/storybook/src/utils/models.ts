export type UiFramework =
  | '@storybook/angular'
  | '@storybook/html-webpack5'
  | '@storybook/nextjs'
  | '@storybook/preact-webpack5'
  | '@storybook/react-webpack5'
  | '@storybook/react-vite'
  | '@storybook/server-webpack5'
  | '@storybook/svelte-webpack5'
  | '@storybook/svelte-vite'
  | '@storybook/sveltekit'
  | '@storybook/vue-webpack5'
  | '@storybook/vue-vite'
  | '@storybook/vue3-webpack5'
  | '@storybook/vue3-vite'
  | '@storybook/web-components-webpack5'
  | '@storybook/web-components-vite';
