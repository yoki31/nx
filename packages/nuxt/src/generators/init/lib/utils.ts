import {
  addDependenciesToPackageJson,
  readNxJson,
  Tree,
  updateNxJson,
} from '@nx/devkit';
import { nuxtVersion, nxVersion } from '../../../utils/versions';
import { InitSchema } from '../schema';

export function updateDependencies(host: Tree, schema: InitSchema) {
  return addDependenciesToPackageJson(
    host,
    {},
    {
      '@nx/nuxt': nxVersion,
      nuxt: nuxtVersion,
      '@nx/vite': nxVersion,
    },
    undefined,
    schema.keepExistingVersions
  );
}

export function addVitestTargetDefaults(tree: Tree) {
  const nxJson = readNxJson(tree);

  const productionFileSet = nxJson.namedInputs?.production;
  if (productionFileSet) {
    productionFileSet.push(
      '!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)',
      '!{projectRoot}/tsconfig.spec.json'
    );

    nxJson.namedInputs.production = Array.from(new Set(productionFileSet));
  }

  updateNxJson(tree, nxJson);
}

export function addPlugin(tree: Tree) {
  const nxJson = readNxJson(tree);
  nxJson.plugins ??= [];

  let hasNxNuxtPlugin = false;
  let hasNxVitePlugin = false;

  for (const plugin of nxJson.plugins) {
    if (
      typeof plugin === 'string'
        ? plugin === '@nx/nuxt/plugin'
        : plugin.plugin === '@nx/nuxt/plugin'
    ) {
      hasNxNuxtPlugin = true;
    }
    if (
      typeof plugin === 'string'
        ? plugin === '@nx/vite/plugin'
        : plugin.plugin === '@nx/vite/plugin'
    ) {
      hasNxVitePlugin = true;
    }
  }

  if (!hasNxNuxtPlugin) {
    nxJson.plugins.push({
      plugin: '@nx/nuxt/plugin',
      options: {
        buildTargetName: 'build',
        serveTargetName: 'serve',
        serveStaticTargetName: 'serve-static',
      },
    });
  }

  if (!hasNxVitePlugin) {
    nxJson.plugins.push({
      plugin: '@nx/vite/plugin',
      options: {
        buildTargetName: 'build',
        previewTargetName: 'preview',
        testTargetName: 'test',
        serveTargetName: 'serve',
        serveStaticTargetName: 'serve-static',
      },
    });
  }

  updateNxJson(tree, nxJson);
}
