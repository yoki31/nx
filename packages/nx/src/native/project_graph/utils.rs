use crate::native::project_graph::types::Project;
use std::collections::HashMap;

mod find_project_for_path;
pub use find_project_for_path::*;

pub type ProjectRootMappings = HashMap<String, String>;
pub fn create_project_root_mappings(nodes: &HashMap<String, Project>) -> ProjectRootMappings {
    let mut project_root_mappings = HashMap::new();
    for (project_name, node) in nodes {
        project_root_mappings.insert(node.root.clone(), normalize_project_root(project_name));
    }
    project_root_mappings
}

pub fn normalize_project_root(root: &str) -> String {
    let root = if root.is_empty() {
        ".".to_string()
    } else {
        root.to_owned()
    };
    if root.ends_with('/') {
        root.strip_suffix('/')
            .expect("'/' already checked to exist")
            .to_string()
    } else {
        root.to_owned()
    }
}
