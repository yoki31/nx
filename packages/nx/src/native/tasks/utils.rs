use crate::native::project_graph::types::ProjectGraph;
use hashbrown::HashSet;

pub(super) fn find_all_project_node_dependencies<'a>(
    parent_node_name: &str,
    project_graph: &'a ProjectGraph,
    exclude_external_dependencies: bool,
) -> Vec<&'a String> {
    let mut dependent_node_names = HashSet::new();
    collect_dependent_project_node_names(
        project_graph,
        parent_node_name,
        &mut dependent_node_names,
        exclude_external_dependencies,
    );
    dependent_node_names.into_iter().collect()
}

fn collect_dependent_project_node_names<'a>(
    project_graph: &'a ProjectGraph,
    parent_node_name: &str,
    dependent_node_names: &mut HashSet<&'a String>,
    exclude_external_dependencies: bool,
) {
    let Some(dependencies) = project_graph.dependencies.get(parent_node_name) else {
        return;
    };

    for dependency in dependencies {
        // skip dependencies already added (avoid circular dependencies)
        if dependent_node_names.contains(dependency) {
            continue;
        }

        if exclude_external_dependencies && project_graph.external_nodes.contains_key(dependency) {
            continue;
        }

        dependent_node_names.insert(dependency);
        collect_dependent_project_node_names(
            project_graph,
            dependency,
            dependent_node_names,
            exclude_external_dependencies,
        );
    }
}
