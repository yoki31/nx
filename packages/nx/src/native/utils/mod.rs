mod find_matching_projects;
mod get_mod_time;
mod normalize_trait;
pub mod path;

pub use find_matching_projects::*;
pub use get_mod_time::*;
pub use normalize_trait::Normalize;
