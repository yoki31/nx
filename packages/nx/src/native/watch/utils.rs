use ignore::WalkBuilder;
use ignore_files::IgnoreFile;
use std::path::Path;
use std::{fs, path::PathBuf};
use tracing::trace;
use watchexec_events::{Event, Tag};

pub(super) fn get_ignore_files<T: AsRef<str>>(
    use_ignore: bool,
    root: T,
) -> Option<Vec<IgnoreFile>> {
    let root = root.as_ref();
    if use_ignore {
        let mut walker = WalkBuilder::new(root);
        walker.hidden(false);
        walker.git_ignore(false);

        let node_folder = PathBuf::from(root).join("node_modules");
        walker.filter_entry(move |entry| !entry.path().starts_with(&node_folder));
        Some(
            walker
                .build()
                .flatten()
                .filter(|result| result.path().ends_with(".gitignore"))
                .map(|result| {
                    let path: PathBuf = result.path().into();
                    let parent: PathBuf = path.parent().unwrap_or(&path).into();
                    IgnoreFile {
                        path,
                        applies_in: Some(parent),
                        applies_to: None,
                    }
                })
                .collect(),
        )
    } else {
        None
    }
}

pub(super) fn get_nx_ignore<P: AsRef<Path>>(origin: P) -> Option<IgnoreFile> {
    let nx_ignore_path = PathBuf::from(origin.as_ref()).join(".nxignore");
    if nx_ignore_path.exists() {
        Some(IgnoreFile {
            path: nx_ignore_path,
            applies_in: Some(origin.as_ref().into()),
            applies_to: None,
        })
    } else {
        None
    }
}

pub(super) fn transform_event(watch_event: &Event) -> Option<Event> {
    if cfg!(target_os = "linux") {
        let tags = watch_event
            .tags
            .clone()
            .into_iter()
            .map(|tag| match tag {
                Tag::Path { path, file_type } => {
                    trace!("canonicalizing {:?}", path);
                    let real_path = fs::canonicalize(&path).unwrap_or(path);
                    trace!("real path {:?}", real_path);
                    Tag::Path {
                        path: real_path,
                        file_type,
                    }
                }
                _ => tag,
            })
            .collect();

        Some(Event {
            tags,
            metadata: watch_event.metadata.clone(),
        })
    } else {
        None
    }
}
