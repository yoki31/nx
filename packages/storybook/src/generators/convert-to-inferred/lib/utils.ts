import { tsquery } from '@phenomnomnominal/tsquery';
import { readJson, joinPathFragments, type Tree } from '@nx/devkit';
import { AggregatedLog } from '@nx/devkit/src/generators/plugin-migrations/aggregate-log-util';
import { toProjectRelativePath } from '@nx/devkit/src/generators/plugin-migrations/plugin-migration-utils';
import { dirname } from 'path/posix';
import { coerce, major } from 'semver';

export function getConfigFilePath(tree: Tree, configDir: string) {
  return [
    joinPathFragments(configDir, `main.ts`),
    joinPathFragments(configDir, `main.cts`),
    joinPathFragments(configDir, `main.mts`),
    joinPathFragments(configDir, `main.js`),
    joinPathFragments(configDir, `main.cjs`),
    joinPathFragments(configDir, `main.mjs`),
  ].find((f) => tree.exists(f));
}

export function addConfigValuesToConfigFile(
  tree: Tree,
  configFile: string,
  configValues: Record<string, Record<string, unknown>>
) {
  const IMPORT_PROPERTY_SELECTOR = 'ImportDeclaration';
  const configFileContents = tree.read(configFile, 'utf-8');

  const ast = tsquery.ast(configFileContents);
  // AST TO GET SECTION TO APPEND TO
  const importNodes = tsquery(ast, IMPORT_PROPERTY_SELECTOR, {
    visitAllChildren: true,
  });
  let startPosition = 0;
  if (importNodes.length !== 0) {
    const lastImportNode = importNodes[importNodes.length - 1];
    startPosition = lastImportNode.getEnd();
  }

  const configValuesString = `
  // These options were migrated by @nx/storybook:convert-to-inferred from the project.json file.
  const configValues = ${JSON.stringify(configValues)};
  
  // Determine the correct configValue to use based on the configuration
  const nxConfiguration = process.env.NX_TASK_TARGET_CONFIGURATION ?? 'default';
  
  const options = {
    ...configValues.default,
    ...(configValues[nxConfiguration] ?? {})
  }`;

  tree.write(
    configFile,
    `${configFileContents.slice(0, startPosition)}
  ${configValuesString}
  ${configFileContents.slice(startPosition)}`
  );
}

export const STORYBOOK_PROP_MAPPINGS = {
  v7: {
    port: 'port',
    previewUrl: 'preview-url',
    host: 'host',
    docs: 'docs',
    configDir: 'config-dir',
    logLevel: 'loglevel',
    quiet: 'quiet',
    webpackStatsJson: 'webpack-stats-json',
    debugWebpack: 'debug-webpack',
    disableTelemetry: 'disable-telemetry',
    https: 'https',
    sslCa: 'ssl-ca',
    sslCert: 'ssl-cert',
    sslKey: 'ssl-key',
    smokeTest: 'smoke-test',
    noOpen: 'no-open',
    outputDir: 'output-dir',
  },
  v8: {
    port: 'port',
    previewUrl: 'preview-url',
    host: 'host',
    docs: 'docs',
    configDir: 'config-dir',
    logLevel: 'loglevel',
    quiet: 'quiet',
    webpackStatsJson: 'stats-json',
    debugWebpack: 'debug-webpack',
    disableTelemetry: 'disable-telemetry',
    https: 'https',
    sslCa: 'ssl-ca',
    sslCert: 'ssl-cert',
    sslKey: 'ssl-key',
    smokeTest: 'smoke-test',
    noOpen: 'no-open',
    outputDir: 'output-dir',
  },
};

export function ensureViteConfigPathIsRelative(
  tree: Tree,
  configPath: string,
  projectName: string,
  projectRoot: string,
  executorName: string,
  migrationLogs: AggregatedLog
) {
  const configFileContents = tree.read(configPath, 'utf-8');

  if (configFileContents.includes('viteFinal:')) {
    return;
  }

  const ast = tsquery.ast(configFileContents);
  const REACT_FRAMEWORK_SELECTOR_IDENTIFIERS =
    'PropertyAssignment:has(Identifier[name=framework]) PropertyAssignment:has(Identifier[name=name]) StringLiteral[value=@storybook/react-vite]';
  const REACT_FRAMEWORK_SELECTOR_STRING_LITERALS =
    'PropertyAssignment:has(StringLiteral[value=framework]) PropertyAssignment:has(StringLiteral[value=name]) StringLiteral[value=@storybook/react-vite]';

  const VUE_FRAMEWORK_SELECTOR_IDENTIFIERS =
    'PropertyAssignment:has(Identifier[name=framework]) PropertyAssignment:has(Identifier[name=name]) StringLiteral[value=@storybook/vue3-vite]';
  const VUE_FRAMEWORK_SELECTOR_STRING_LITERALS =
    'PropertyAssignment:has(StringLiteral[value=framework]) PropertyAssignment:has(StringLiteral[value=name]) StringLiteral[value=@storybook/vue3-vite]';
  const isUsingVite =
    tsquery(ast, REACT_FRAMEWORK_SELECTOR_IDENTIFIERS, {
      visitAllChildren: true,
    }).length > 0 ||
    tsquery(ast, REACT_FRAMEWORK_SELECTOR_STRING_LITERALS, {
      visitAllChildren: true,
    }).length > 0 ||
    tsquery(ast, VUE_FRAMEWORK_SELECTOR_STRING_LITERALS, {
      visitAllChildren: true,
    }).length > 0 ||
    tsquery(ast, VUE_FRAMEWORK_SELECTOR_IDENTIFIERS, { visitAllChildren: true })
      .length > 0;
  if (!isUsingVite) {
    return;
  }

  const VITE_CONFIG_PATH_SELECTOR =
    'PropertyAssignment:has(Identifier[name=framework]) PropertyAssignment PropertyAssignment PropertyAssignment:has(Identifier[name=viteConfigPath]) > StringLiteral';
  let viteConfigPathNodes = tsquery(ast, VITE_CONFIG_PATH_SELECTOR, {
    visitAllChildren: true,
  });
  if (viteConfigPathNodes.length === 0) {
    const VITE_CONFIG_PATH_SELECTOR_STRING_LITERALS =
      'PropertyAssignment:has(StringLiteral[value=framework]) PropertyAssignment PropertyAssignment PropertyAssignment:has(StringLiteral[value=viteConfigPath]) > StringLiteral:not(StringLiteral[value=viteConfigPath])';
    viteConfigPathNodes = tsquery(
      ast,
      VITE_CONFIG_PATH_SELECTOR_STRING_LITERALS,
      {
        visitAllChildren: true,
      }
    );

    if (viteConfigPathNodes.length === 0) {
      migrationLogs.addLog({
        project: projectName,
        executorName,
        log: 'Unable to find `viteConfigPath` in Storybook Config. Please ensure the `viteConfigPath` is relative to the project root.',
      });
      return;
    }
  }

  const viteConfigPathNode = viteConfigPathNodes[0];
  const pathToViteConfig = viteConfigPathNode.getText().replace(/('|")/g, '');
  if (pathToViteConfig.match(/^(\.\.\/|\.\/)/)) {
    return;
  }
  const relativePathToViteConfig = toProjectRelativePath(
    pathToViteConfig,
    projectRoot
  );

  tree.write(
    configPath,
    `${configFileContents.slice(
      0,
      viteConfigPathNode.getStart() + 1
    )}${relativePathToViteConfig}${configFileContents.slice(
      viteConfigPathNode.getEnd() - 1
    )}`
  );
}

export function getInstalledPackageVersion(
  tree: Tree,
  pkgName: string
): string | null {
  const { dependencies, devDependencies } = readJson(tree, 'package.json');
  const version = dependencies?.[pkgName] ?? devDependencies?.[pkgName];

  return version;
}

export function getInstalledPackageVersionInfo(tree: Tree, pkgName: string) {
  const version = getInstalledPackageVersion(tree, pkgName);

  return version ? { major: major(coerce(version)), version } : null;
}
