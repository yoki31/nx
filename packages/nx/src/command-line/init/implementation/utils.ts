import { execSync } from 'child_process';
import { join } from 'path';

import { NxJsonConfiguration } from '../../../config/nx-json';
import { runNxSync } from '../../../utils/child-process';
import {
  fileExists,
  readJsonFile,
  writeJsonFile,
} from '../../../utils/fileutils';
import { output } from '../../../utils/output';
import { PackageJson } from '../../../utils/package-json';
import {
  getPackageManagerCommand,
  PackageManagerCommands,
} from '../../../utils/package-manager';
import { joinPathFragments } from '../../../utils/path';
import { nxVersion } from '../../../utils/versions';
import { existsSync, readFileSync, writeFileSync } from 'fs';

export function createNxJsonFile(
  repoRoot: string,
  topologicalTargets: string[],
  cacheableOperations: string[],
  scriptOutputs: { [name: string]: string }
) {
  const nxJsonPath = joinPathFragments(repoRoot, 'nx.json');
  let nxJson = {} as Partial<NxJsonConfiguration> & { $schema: string };
  try {
    nxJson = readJsonFile(nxJsonPath);
    // eslint-disable-next-line no-empty
  } catch {}

  nxJson.$schema = './node_modules/nx/schemas/nx-schema.json';
  nxJson.targetDefaults ??= {};

  if (topologicalTargets.length > 0) {
    for (const scriptName of topologicalTargets) {
      nxJson.targetDefaults[scriptName] ??= {};
      nxJson.targetDefaults[scriptName] = { dependsOn: [`^${scriptName}`] };
    }
  }
  for (const [scriptName, output] of Object.entries(scriptOutputs)) {
    if (!output) {
      // eslint-disable-next-line no-continue
      continue;
    }
    nxJson.targetDefaults[scriptName] ??= {};
    nxJson.targetDefaults[scriptName].outputs = [`{projectRoot}/${output}`];
  }

  for (const target of cacheableOperations) {
    nxJson.targetDefaults[target] ??= {};
    nxJson.targetDefaults[target].cache ??= true;
  }

  if (Object.keys(nxJson.targetDefaults).length === 0) {
    delete nxJson.targetDefaults;
  }

  nxJson.defaultBase ??= deduceDefaultBase();
  writeJsonFile(nxJsonPath, nxJson);
}

function deduceDefaultBase() {
  try {
    execSync(`git rev-parse --verify main`, {
      stdio: ['ignore', 'ignore', 'ignore'],
    });
    return 'main';
  } catch {
    try {
      execSync(`git rev-parse --verify dev`, {
        stdio: ['ignore', 'ignore', 'ignore'],
      });
      return 'dev';
    } catch {
      try {
        execSync(`git rev-parse --verify develop`, {
          stdio: ['ignore', 'ignore', 'ignore'],
        });
        return 'develop';
      } catch {
        try {
          execSync(`git rev-parse --verify next`, {
            stdio: ['ignore', 'ignore', 'ignore'],
          });
          return 'next';
        } catch {
          return 'master';
        }
      }
    }
  }
}

export function addDepsToPackageJson(
  repoRoot: string,
  additionalPackages?: string[]
) {
  const path = joinPathFragments(repoRoot, `package.json`);
  const json = readJsonFile(path);
  if (!json.devDependencies) json.devDependencies = {};
  json.devDependencies['nx'] = nxVersion;
  if (additionalPackages) {
    for (const p of additionalPackages) {
      json.devDependencies[p] = nxVersion;
    }
  }
  writeJsonFile(path, json);
}

export function updateGitIgnore(root: string) {
  const ignorePath = join(root, '.gitignore');
  try {
    let contents = readFileSync(ignorePath, 'utf-8');
    const lines = contents.split('\n');
    let sepIncluded = false;
    if (!contents.includes('.nx/cache')) {
      if (!sepIncluded) {
        lines.push('\n');
        sepIncluded = true;
      }
      lines.push('.nx/cache');
    }
    if (!contents.includes('.nx/workspace-data')) {
      if (!sepIncluded) {
        lines.push('\n');
        sepIncluded = true;
      }
      lines.push('.nx/workspace-data');
    }
    writeFileSync(ignorePath, lines.join('\n'), 'utf-8');
  } catch {}
}

export function runInstall(
  repoRoot: string,
  pmc: PackageManagerCommands = getPackageManagerCommand()
) {
  execSync(pmc.install, { stdio: [0, 1, 2], cwd: repoRoot });
}

export function initCloud(
  repoRoot: string,
  installationSource:
    | 'nx-init-angular'
    | 'nx-init-cra'
    | 'nx-init-monorepo'
    | 'nx-init-nest'
    | 'nx-init-npm-repo'
) {
  runNxSync(
    `g nx:connect-to-nx-cloud --installationSource=${installationSource} --quiet --no-interactive`,
    {
      stdio: [0, 1, 2],
      cwd: repoRoot,
    }
  );
}

export function addVsCodeRecommendedExtensions(
  repoRoot: string,
  extensions: string[]
): void {
  const vsCodeExtensionsPath = join(repoRoot, '.vscode/extensions.json');

  if (fileExists(vsCodeExtensionsPath)) {
    const vsCodeExtensionsJson = readJsonFile(vsCodeExtensionsPath);

    vsCodeExtensionsJson.recommendations ??= [];
    extensions.forEach((extension) => {
      if (!vsCodeExtensionsJson.recommendations.includes(extension)) {
        vsCodeExtensionsJson.recommendations.push(extension);
      }
    });

    writeJsonFile(vsCodeExtensionsPath, vsCodeExtensionsJson);
  } else {
    writeJsonFile(vsCodeExtensionsPath, { recommendations: extensions });
  }
}

export function markRootPackageJsonAsNxProjectLegacy(
  repoRoot: string,
  cacheableScripts: string[],
  pmc: PackageManagerCommands
) {
  const json = readJsonFile<PackageJson>(
    joinPathFragments(repoRoot, `package.json`)
  );
  json.nx = {};
  for (let script of cacheableScripts) {
    const scriptDefinition = json.scripts[script];
    if (!scriptDefinition) {
      continue;
    }

    if (scriptDefinition.includes('&&') || scriptDefinition.includes('||')) {
      let backingScriptName = `_${script}`;
      json.scripts[backingScriptName] = scriptDefinition;
      json.scripts[script] = `nx exec -- ${pmc.run(backingScriptName, '')}`;
    } else {
      json.scripts[script] = `nx exec -- ${json.scripts[script]}`;
    }
  }
  writeJsonFile(`package.json`, json);
}

export function markPackageJsonAsNxProject(packageJsonPath: string) {
  const json = readJsonFile<PackageJson>(packageJsonPath);
  if (!json.scripts) {
    return;
  }

  json.nx = {};
  writeJsonFile(packageJsonPath, json);
}

export function printFinalMessage({
  learnMoreLink,
}: {
  learnMoreLink?: string;
}): void {
  const pmc = getPackageManagerCommand();

  output.success({
    title: 'ðŸŽ‰ Done!',
    bodyLines: [
      `- Run "${pmc.exec} nx run-many -t build" to run the build target for every project in the workspace. Run it again to replay the cached computation. https://nx.dev/features/cache-task-results`,
      `- Run "${pmc.exec} nx graph" to see the graph of projects and tasks in your workspace. https://nx.dev/core-features/explore-graph`,
      learnMoreLink ? `- Learn more at ${learnMoreLink}.` : undefined,
    ].filter(Boolean),
  });
}

export function isMonorepo(packageJson: PackageJson) {
  if (!!packageJson.workspaces) return true;

  if (existsSync('pnpm-workspace.yaml') || existsSync('pnpm-workspace.yml'))
    return true;

  if (existsSync('lerna.json')) return true;

  return false;
}
