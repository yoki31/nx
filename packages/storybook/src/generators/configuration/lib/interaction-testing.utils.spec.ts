import { ProjectConfiguration, Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { addInteractionsInAddons } from './interaction-testing.utils';
describe('Helper functions for the Storybook 7 migration generator', () => {
  let tree: Tree;

  beforeEach(async () => {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  });

  it('should add addon-interactions in main.ts if it does not exist', () => {
    tree.write(
      `.storybook/main.ts`,
      `import type { StorybookConfig } from '@storybook/angular';
    
            const config: StorybookConfig = {
              stories: ['../**/*.stories.@(js|jsx|ts|tsx|mdx)'],
              addons: ['@storybook/addon-essentials'],
              framework: {
                name: '@storybook/angular',
                options: {},
              },
            };
            
            export default config;`
    );
    addInteractionsInAddons(tree, {
      name: 'my-proj',
      targets: {
        storybook: {
          executor: '@nx/storybook:storybook',
          options: {
            configDir: `.storybook`,
          },
        },
      },
    } as unknown as ProjectConfiguration);
    expect(tree.read(`.storybook/main.ts`, 'utf-8')).toMatchSnapshot();
  });

  it('should do nothing if addon-interactions already exists in main.ts', () => {
    tree.write(
      `.storybook/main.ts`,
      `import type { StorybookConfig } from '@storybook/angular';
    
            const config: StorybookConfig = {
              stories: ['../**/*.stories.@(js|jsx|ts|tsx|mdx)'],
              addons: ['@storybook/addon-essentials', '@storybook/addon-interactions'],
              framework: {
                name: '@storybook/angular',
                options: {},
              },
            };
            
            export default config;`
    );
    addInteractionsInAddons(tree, {
      name: 'my-proj',
      targets: {
        storybook: {
          executor: '@nx/storybook:storybook',
          options: {
            configDir: `.storybook`,
          },
        },
      },
    } as unknown as ProjectConfiguration);
    expect(tree.read(`.storybook/main.ts`, 'utf-8')).toMatchSnapshot();
  });
});
