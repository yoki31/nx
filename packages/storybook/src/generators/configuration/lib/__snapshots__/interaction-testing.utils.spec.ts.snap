// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helper functions for the Storybook 7 migration generator should add addon-interactions in main.ts if it does not exist 1`] = `
"import type { StorybookConfig } from '@storybook/angular';
    
            const config: StorybookConfig = {
              stories: ['../**/*.stories.@(js|jsx|ts|tsx|mdx)'],
              addons: ['@storybook/addon-interactions', '@storybook/addon-essentials'],
              framework: {
                name: '@storybook/angular',
                options: {},
              },
            };
            
            export default config;"
`;

exports[`Helper functions for the Storybook 7 migration generator should do nothing if addon-interactions already exists in main.ts 1`] = `
"import type { StorybookConfig } from '@storybook/angular';
    
            const config: StorybookConfig = {
              stories: ['../**/*.stories.@(js|jsx|ts|tsx|mdx)'],
              addons: ['@storybook/addon-essentials', '@storybook/addon-interactions'],
              framework: {
                name: '@storybook/angular',
                options: {},
              },
            };
            
            export default config;"
`;
